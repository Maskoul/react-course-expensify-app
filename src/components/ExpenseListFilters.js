import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, setSortByDate, setSortByAmount, setStartDate, setEndDate } from '../actions/filters';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)

    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}))
    };
    onTextChange = (e) => this.props.setTextFilter(e.target.value);

    onSortByChange = (e) => {
        if(e.target.value === 'date'){
            this.props.setSortByDate()
        }else if(e.target.value === 'amount'){
            this.props.setSortByAmount()
        }
    };
    render() {
        return (
            <div>
                <input 
                type= 'text'
                value= {this.props.filters.text}
                onChange= {this.onTextChange}
                />
                <select 
                    value={this.props.filters.sortBy}
                    onChange={this.onSortByChange}>
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>

                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    startDateId={"1"}
                    endDate={this.props.filters.startDate}
                    endDateId={"2"}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    showClearDates={true}
                    isOutsideRange={() => false}
                />
            </div>
        );
    }
}
const mapStateToProps = (state) =>({
        filters: state.filters

});

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    setSortByDate: () => dispatch(setSortByDate()),
    setSortByAmount:() => dispatch(setSortByAmount())
});
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);