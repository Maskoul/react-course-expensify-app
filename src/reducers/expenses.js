const defaultExpensesReducer = []

export default (state = defaultExpensesReducer, action) => {
    switch(action.type){
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];  
        case 'REMOVE_EXPENSE': 
               return state.filter((expense) => {
                    if(action.id !== expense.id){
                        return expense
                    }
                });

        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if(expense.id === action.id){
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense;
                }
            });
        default:
            return  state
    }
};