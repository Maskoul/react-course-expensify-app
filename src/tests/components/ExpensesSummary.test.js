import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expense';

test('should render ExpensesSummary without data correctly',() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={250} />);
    expect(wrapper).toMatchSnapshot();
});


test('should render ExpensesSummary with data correctly',() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={55} expensesTotal={52062331325}/>);
    expect(wrapper).toMatchSnapshot();
});
