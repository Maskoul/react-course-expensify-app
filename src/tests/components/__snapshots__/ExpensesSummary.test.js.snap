// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with data correctly 1`] = `
<div>
  <h1>
    Viewing 
    55
     
    expenses
     totalling 
    $520,623,313.25
  </h1>
</div>
`;

exports[`should render ExpensesSummary without data correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.50
  </h1>
</div>
`;
